src/util_test.go

```diff
@@ -9,76 +9,120 @@ import (
+	testCases := []struct {
+		name          string
+		stdout        string
+		correlationID string
+		contentType   string
+	}{
+		{
+			name:          "Output file contains expected data",
+			stdout:        "Hello, world!",
+			correlationID: "12345",
+			contentType:   "application/json",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			body, boundary := getOutputFile(tc.stdout, tc.correlationID, tc.contentType)
+
+			// Verify that the boundary is not empty
+			if boundary == "" {
+				t.Error("Boundary should not be empty")
+			}
+
+			expectedPayload := fmt.Sprintf(`{"correlation_id":"%s","stdout":"%s"}`, tc.correlationID, tc.stdout)
+			gotTrimmed := strings.TrimSpace(body.String())
+
+			// Verify that the body contains the expected data
+			if !strings.Contains(gotTrimmed, expectedPayload) {
+				t.Errorf("Unexpected body payload. Expected to contain: %s, Got: %s", expectedPayload, body.String())
+			}
+			prefix := fmt.Sprintf(`--%s`, boundary)
+			if !strings.HasPrefix(gotTrimmed, prefix) {
+				t.Errorf("Unexpected body payload. Expected to have prefix: %s, Got: %s", prefix, body.String())
+			}
+			suffix := fmt.Sprintf(`--%s--`, boundary)
+			if !strings.HasSuffix(gotTrimmed, suffix) {
+				t.Errorf("Unexpected body payload. Expected to have suffix: %s, Got: %s", suffix, gotTrimmed)
+			}
+			// TODO: test that content type is also there
+		})
 	}
-	prefix := fmt.Sprintf(`--%s`, boundary)
-	if !strings.HasPrefix(gotTrimmed, prefix) {
-		t.Errorf("Unexpected body payload. Expected to have prefix: %s, Got: %s", prefix, body.String())
-	}
-	suffix := fmt.Sprintf(`--%s--`, boundary)
-	if !strings.HasSuffix(gotTrimmed, suffix) {
-		t.Errorf("Unexpected body payload. Expected to have suffix: %s, Got: %s", suffix, gotTrimmed)
-	}
-	// TODO: test that content type is also there
 }
 
 func TestWriteFileToTemporaryDir(t *testing.T) {
-	// Create a temporary directory for testing
-	tempDirPath := "test-dir"
-	defer os.RemoveAll(tempDirPath)
-
-	data := []byte("test data")
-	filePath := writeFileToTemporaryDir(data, tempDirPath)
-
-	// Assert that the file exists
-	_, err := os.Stat(filePath)
-	if err != nil {
-		t.Errorf("Expected file to be created, got error: %v", err)
-	}
-
-	// Assert that the file contains the expected data
-	fileContent, err := os.ReadFile(filePath)
-	if err != nil {
-		t.Errorf("Failed to read file content: %v", err)
-	}
-	if string(fileContent) != string(data) {
-		t.Errorf("Expected file content: %s, got: %s", string(data), string(fileContent))
+	testCases := []struct {
+		name     string
+		data     []byte
+		tempDir  string
+		expected string
+	}{
+		{
+			name:     "File is created and contains expected data",
+			data:     []byte("test data"),
+			tempDir:  "test-dir",
+			expected: "test data",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			// Create a temporary directory for testing
+			tempDirPath := tc.tempDir
```

Comment:
Can't we use the `t.tempDir` instead of a user-defined directory?

Reply:
I didn't know something like that exists :) replaced